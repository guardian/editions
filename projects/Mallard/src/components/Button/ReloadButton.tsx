import React from 'react';
import { View } from 'react-native';
import { useInsets } from 'src/hooks/use-screen';
import { Button, ButtonAppearance } from './Button';

export const ReloadButton: React.FC<{
	onPress: () => void;
}> = ({ onPress }) => {
	const { top, left } = useInsets();
	return (
		<View
			style={[
				{
					position: 'absolute',
					top: top + 20,
					right: 70,
					zIndex: 99999,
				},
			]}
		>
			<Button
				appearance={ButtonAppearance.Tomato}
				onPress={onPress}
				buttonStyles={{ left: 0 }}
			>
				Reload
			</Button>
		</View>
	);
};
