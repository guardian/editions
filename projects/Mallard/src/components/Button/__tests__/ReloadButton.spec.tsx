import React from 'react';
import type { ReactTestRendererJSON } from 'react-test-renderer';
import TestRenderer from 'react-test-renderer';
import { ReloadButton } from '../ReloadButton';

describe('ReloadButton', () => {
	it('should display a ReloadButton', () => {
		const component: ReactTestRendererJSON | null = TestRenderer.create(
			<ReloadButton onPress={() => {}} />,
		).toJSON();
		expect(component).toMatchSnapshot();
	});
});
