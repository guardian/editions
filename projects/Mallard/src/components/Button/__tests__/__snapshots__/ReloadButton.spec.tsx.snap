// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadButton should display a ReloadButton 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
        "right": 70,
        "top": 40,
        "zIndex": 99999,
      },
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 999,
            "height": 42,
            "justifyContent": "center",
            "paddingHorizontal": 21,
          },
          Object {
            "backgroundColor": "#FF5943",
          },
          false,
          Object {
            "left": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "flexShrink": 0,
              "fontFamily": "GuardianTextSans-Regular",
              "fontSize": 17,
              "lineHeight": 21,
            },
            Object {
              "color": "#121212",
            },
            Array [
              Object {
                "flexShrink": 0,
                "fontFamily": "GuardianTextSans-Bold",
                "fontSize": 17,
                "lineHeight": 21,
              },
              Object {
                "textAlign": "auto",
              },
              Object {
                "color": "#ffffff",
              },
              undefined,
            ],
          ]
        }
      >
        Reload
      </Text>
    </View>
  </View>
</View>
`;
